kql/kql2eql.py

```diff
@@ -51,7 +53,12 @@ def field_value_expression(self, tree):
 
         with self.scope(self.visit(field_tree)) as field_name:
             # check the field against the schema
-            self.get_field_type(field_name, field_tree)
+
+            type_mapping = self.get_field_type(field_name, field_tree)
```

Comment:
Adding it here will only catch it when converting KQL to EQL, so this error would not be raised with direct eql schema validation used within the rule objects (`KqlSchema2Eql`). 

The problem that arises though, is that this failure occurs in the parser, which consults the schema `get_event_type_hint` method which we _do_ overload.


https://github.com/endgameinc/eql/blob/535bcdf5f7ac5308c8b7c0045cd428140136d9e9/eql/parser.py#L443

https://github.com/elastic/detection-rules/blob/2ed97d2e8c38fe776b16a316a2d048cdb0ec52ca/detection_rules/ecs.py#L194

So to do it correctly, we would need to overload the parser as well, but that doesn't make a lot of sense here. So we _could_ catch it in the overloaded `get_event_type_hint` method, but that also is a bit of a stray away from the design of eql.

I believe this is why we left `text` out from:
https://github.com/elastic/detection-rules/blob/2ed97d2e8c38fe776b16a316a2d048cdb0ec52ca/detection_rules/ecs.py#L174-L183

As a _soft_ way to fail on text fields.

So ultimately, the best place to really implement this for more verbosity would be in the eql validator

https://github.com/elastic/detection-rules/blob/2ed97d2e8c38fe776b16a316a2d048cdb0ec52ca/detection_rules/rule_validators.py#L96

Then just inspect the `exc` and if it is a `text` field, update the message to be more specific from `exc.error_msg`

Reply:
>Then just inspect the exc and if it is a text field, update the message to be more specific from exc.error_msg

Even that is not great, since we lose the context of the bad field with the exc

Reply:
Could just leave it generic-ish like

>                if "Unknown field" in message and beat_types:
                    trailer = f"\nTry adding event.module or event.dataset to specify beats module\n\n{trailer}"

and go 

```python
                if exc.error_msg.startswith('Field not recognized'):
                    trailer = f"\nAre you referencing any unsupported text fields?
```

Reply:
Just seeing this. I went with something more explicit, but I can push this more generic error message if you think its better.